src/main/java/com/b44t/messenger/DcContext.java

```diff
@@ -236,6 +236,19 @@ public boolean isChatmail() {
       return getConfigInt("is_chatmail") == 1;
     }
 
+    // Called for new profiles on chatmail servers that are "single device" initially;
+    // to save server disk space, we make use of that delete all messages immediately after download.
+    public void assumeSingleDevice() {
+      setConfigInt("delete_server_after", 1 /*at once*/);
+    }
+
+    // Called when we get a hint that another device may be set up.
+    public void assumeMultiDevice() {
+      if (getConfigInt("delete_server_after") == 1 /*at once*/) {
```

Comment:
```suggestion
      if (isChatmail() && getConfigInt("delete_server_after") == 1 /*at once*/) {
```

that might be better - but requires `is_chatmail` to be set accordingly after backup import or on export when no network is available - afaik, this is the case, but not 100% sure

Reply:
did an explict commit chat checks for is_chatmail on both, assumeSingleDevice() as well as assumeMultiDevice()